commcare/src/main/java/org/motechproject/commcare/tasks/builder/CaseActionBuilder.java

```diff
@@ -75,16 +85,32 @@ public CaseActionBuilder(CommcareConfigService configService) {
                     .setOrder(order++);
             parameters.add(builder.createActionParameterRequest());
 
-            String displayName = DisplayNameHelper.buildDisplayName(DisplayNames.CREATE_CASE, config.getName());
+            order = 0;
+            builder = new ActionParameterRequestBuilder()
+                    .setDisplayName(DisplayNames.CASE_ID)
+                    .setKey(EventDataKeys.CASE_ID)
+                    .setRequired(false)
+                    .setOrder(order++);
+            postActionParameters.add(builder.createActionParameterRequest());
+
+            String displayName = DisplayNameHelper.buildDisplayName(DisplayNames.CREATE_CASE, configName);
             ActionEventRequestBuilder actionBuilder = new ActionEventRequestBuilder()
                     .setDisplayName(displayName)
-                    .setSubject(EventSubjects.CREATE_CASE + "." + config.getName())
-                    .setActionParameters(parameters);
+                    .setServiceInterface(ACTION_CASE_SERVICE)
+                    .setServiceMethod(serviceMethod)
+                    .setSubject(EventSubjects.CREATE_CASE + "." + configName)
```

Comment:
I checked in code. If action's service is set and it is available, event won't be sent and only service method will be invoked.